Shared/funcannotations.h

```diff
@@ -47,6 +47,9 @@
 #if defined(__CUDACC__) || (defined(__GNUC__) && defined(__SANITIZE_THREAD__)) || \
```

Comment:
These modes are currently mutually exclusive. The attribute is necessary for the proper processing of "external always_inline" functions in case of shared library build.